### PR TITLE
Add unsupported config override for config manager

### PR DESCRIPTION
**Pull request by Miciah**
Add an unsupported config override to enable OpenShift router's dynamic config manager for HAProxy.

* `pkg/operator/controller/ingress/deployment.go` (`RouterHAProxyConfigManager`): New const.
(`desiredRouterDeployment`): Add unsupported config override for `ROUTER_HAPROXY_CONFIG_MANAGER`.
* `pkg/operator/controller/ingress/deployment_test.go` (`TestDesiredRouterDeployment`): Verify that `desiredRouterDeployment` sets `ROUTER_HAPROXY_CONFIG_MANAGER` as expected.
* `test/e2e/operator_test.go` (`TestDynamicConfigManagerUnsupportedConfigOverride`): Verify that the operator sets `ROUTER_HAPROXY_CONFIG_MANAGER` on a router deployment for an IngressController that enables the unsupported config override.

### CODE REVIEW

**Comment by Miciah** [comment]
Rebased.

File: test/e2e/operator_test.go [hunk 1]
```diff
@@ -1985,6 +1985,44 @@ func TestLoadBalancingAlgorithmUnsupportedConfigOverride(t *testing.T) {
...
+	}
+
+	ic.Spec.UnsupportedConfigOverrides = runtime.RawExtension{
+		Raw: []byte(`{"dynamicConfigManager":"true"}`),
+	}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+	if err := waitForDeploymentEnvVar(t, kclient, deployment, 1*time.Minute, "ROUTER_HAPROXY_CONFIG_MANAGER", "true"); err != nil {
+		t.Fatalf("expected initial deployment to set ROUTER_HAPROXY_CONFIG_MANAGER=true: %v", err)
```
Comment (frobware) [inline test/e2e/operator_test.go:2022]: "expected updated deployment"?

Comment (Miciah) [reply test/e2e/operator_test.go:2022]: Fixed.  Thanks!

**Comment by showuon** [comment]
@Miciah , we're wondering for this feature, other than this being explicitly unsupported, are the any drawbacks as to why we should not consider using this a way to address our issues with connection termination? And when it will be officially supported? Thanks.

**Comment by showuon** [comment]
@Miciah @frobware , could you help answer the questions above when you're available? Since we're planning to adopt this dynamic config manager feature in our production environment, we'd really like to get your suggestions about it. Thank you!

**Comment by frobware** [comment]
> @Miciah , we're wondering for this feature, other than this being explicitly unsupported, are the any drawbacks as to why we should not consider using this a way to address our issues with connection termination? And when it will be officially supported? Thanks.

Enabling the override is not currently tested either manually or in any CI jobs that we have/use. I cannot vouch for its current behaviour.

**Comment by showuon** [comment]
@frobware , thanks for the reply. So do you have plan when this feature will be "supported"?

**Comment by frobware** [comment]
@showuon the RFE is tracked here: https://issues.redhat.com/browse/RFE-1439. Notably the spike https://issues.redhat.com/browse/NE-870 is still to be done. I don't have any answer for you as to "when'.

**Comment by showuon** [comment]
Thank you for the comments. I see. Thanks.

**Pull request merged**